Quality_Declaration_fastrtps.md

```diff
@@ -0,0 +1,135 @@
+
+###  Documentation Policy [2.v]
+    
+Changes are not documented on each software change request, although the changes explicitly state what’s fixed/changed, there is no clear consistency among how to document these changes. This is better reflected in releases summaries, where the main new features are shown with the list bugs/issues fixed.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+    
+A complete manual with the whole overview of how the Fast RTPS protocol works, and how the implementation can be used, is provided [here](https://fast-rtps.docs.eprosima.com/en/latest/).
+
+### Public API Documentation [3.ii]
+    
+The public API is documented through their generated [Doxygen documentation](http://www.eprosima.com/docs/fast-rtps/1.5.0/html/group___f_a_s_t_r_t_p_s___g_e_n_e_r_a_l___a_p_i.html) and also in the [User Manual](https://github.com/eProsima/Fast-RTPS/blob/master/doc/pdf/Fast%20RTPS%20-%20User%20Manual.odt) document provided in their repository.
+
+### License [3.iii]
+
+The declared License for the code in the repository is [Apache License 2.0](https://github.com/eProsima/Fast-RTPS/blob/master/LICENSE).
+
+### Copyright Statements [3.iv]
+
+A check to the source files on their repository shows that the source files include copyright statements to *“Proyectos y Sistemas de Mantenimiento SL (eProsima)”*. It is not stated if this is enforced with any kind of linter analysis.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+    
+There are several [tests](https://github.com/eProsima/Fast-RTPS/tree/master/test) in the `fastrtps` repository, the unit tests folder and the communication folders seem to be testing all the features of the software.
+
+### Public API Testing [4.ii]
+    
+There are several [tests](https://github.com/eProsima/Fast-RTPS/tree/master/test) in the `fastrtps` repository, the unit tests folder, the communication, the profiling, and the realtime folders seem to be testing all the public API of the software.
+
+### Coverage [4.iii]
+    
+Checking their [CI jobs](http://jenkins.eprosima.com:8080/), and the testing folders, there seems to be no automatic code coverage analysis being made or anything related to code coverage analysis.
+
+Coverage checks should be made to consider this dependency to apply for a higher quality level.
+
+### Performance [4.iv]
+    
+Fast RTPS includes a directory of performance tests in their [test directory](https://github.com/eProsima/Fast-RTPS/tree/master/test). Some [CI jobs](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Plots/) relating to performance analysis have been run. Based on their frequency, it is unlikely there is a performance regression policy.
```

Comment:
Add a link to the performance test in the buildfarm -> http://build.ros2.org/view/Fci/job/Fci__nightly-performance_ubuntu_focal_amd64/